_posts/en/newsletters/2020-06-24-newsletter.md

```diff
@@ -0,0 +1,238 @@
+  script to trustlessly pay a third party for the final signature needed
+  to complete an adaptor signature.
+
+- **Reminder about collision attack risks on two-party ECDSA:**
+  cryptographer Jonas Nick [replied][nick collision] to the Bitcoin-Dev
+  mailing list thread about a proposed CoinSwap implementation (see
+  [Newsletter #100][news100 coinswap]) reminding developers that P2PKH,
+  P2WPKH, and P2SH addresses which use the 160-bit RIPEMD160 hash are
+  vulnerable to collision attacks that reduce its security to 80 bits
+  when multiple parties collaborate to create an address using a naive
+  protocol (see our [description of this weakness][address security]
+  in legacy P2SH addresses).  Although this was previously only a
+  concern to users of P2SH multisig, it applies in contexts, such as CoinSwap, where
+  it's proposed that two users could share a P2PKH or P2WPKH address.
+  It's possible to avoid this problem, but it requires that the
+  two-party ECDSA protocol be designed to include an extra commitment
+  procedure, which Nick notes some two-party ECDSA protocols and
+  implementations already do.
+
+## Selected Q&A from Bitcoin StackExchange
+
+*[Bitcoin StackExchange][bitcoin.se] is one of the first places Optech
+contributors look for answers to their questions---or when we have a
+few spare moments to help curious or confused users.  In
+this monthly feature, we highlight some of the top-voted questions and
+answers posted since our last update.*
+
+{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aanswer -->{%
+endcomment %}
+{% assign bse = "https://bitcoin.stackexchange.com/a/" %}
+
+- [Why was the current formula that calculates 'target' from 'nBits' chosen?]({{bse}}96298)
+  Ravi Patel asks why a simpler formula for calculating the difficulty target
+  from `nBits` wasn’t chosen. Andrew Chow dives into some of the details around
+  the formula, its history, and even sample code from Bitcoin’s 0.1.5 release.
+
+- [Does Bitcoin really need timestamps?]({{bse}}96185)
+  Pieter Wuille explains why limiting the block rate without reference to a
+  clock time outside of the blockchain could make running full nodes more
+  expensive while also struggling to keep the stale block rate low and
+  preventing collusion attacks.
+
+- [In a fee overpayment attack, why can't compromised software provide fake previous transactions corresponding to fake inputs?]({{bse}}96309)
+  Regarding a [fee overpayment attack][news101 fee overpayment attack] on segwit
+  transactions with multiple inputs, justinmoon asks why the remedy of the
+  attack, requiring copies of previous transactions for the inputs, is not
+  vulnerable to malicious software providing fake previous transactions. Since
+  any provided previous transaction must have a hash that matches the spending input’s
+  previous transaction hash, such an attack is not feasible.
+
+## Releases and release candidates
```

Comment:
Should we mention the HWI release candidate? https://github.com/bitcoin-core/HWI/releases/tag/1.1.2-rc.2

Reply:
I currently only list releases and RCs from the projects listed in the notable commits section, which doesn't currently include HWI.  I had never thought about including HWI in that list until you mentioned it---I guess I never considered it when HWI was a new and experimental project, but now that it's used by several other programs, it's become a legit infrastructure project.

Let's skip an announcement for this week---it's just an RC---but does anyone have any objections to adding HWI to the list of projects we track for notable merges starting next week?

Reply:
:+1: for HWI coverage.